sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/InstrumentationRegistryHelper.java

```diff
@@ -6,22 +6,45 @@
 package com.microsoft.appcenter.utils;
 
 import android.os.Bundle;
-import android.support.test.InstrumentationRegistry;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
 
 /**
  * Wraps InstrumentationRegistry class to enable mocking in unit tests.
  */
 public class InstrumentationRegistryHelper {
 
+    private static final List<String> LOCATIONS = Arrays.asList("androidx.test.platform.app.InstrumentationRegistry",
+            "androidx.test.InstrumentationRegistry",
+            "android.support.test.InstrumentationRegistry");
+
     /**
      * Get the instrumentation arguments from the InstrumentationRegistry. Wrapper exists for unit
      * tests.
      *
      * @return the instrumentation arguments.
      * @throws LinkageError          if the class, method is not found or does not match, typically no test dependencies in release.
-     * @throws IllegalStateException if no argument Bundle has been registered.
      */
     public static Bundle getArguments() throws LinkageError, IllegalStateException {
-        return InstrumentationRegistry.getArguments();
+        Iterator<String> iterator = LOCATIONS.iterator();
+        while (iterator.hasNext()) {
+            String location = iterator.next();
+
+            try {
+                Class<?> aClass = Class.forName(location);
+                Method getArguments = aClass.getMethod("getArguments", (Class[]) null);
+                return (Bundle) getArguments.invoke(null, (Object[]) null);
+            } catch (IllegalStateException | LinkageError e) {
+                if (!iterator.hasNext()) {
+                    throw e;
+                }
+            } catch (Exception ignored) {
+            }
+        }
+        return new Bundle();
```

Comment:
Should actually not return an empty bundle if we could not query any test facility it must print this message: https://github.com/microsoft/appcenter-sdk-android/blob/develop/sdk/appcenter/src/main/java/com/microsoft/appcenter/ServiceInstrumentationUtils.java#L54

Suggestion: simplify code by using Java "foreach", assign exception to a local variable. If we exit for loop, throw that exception instead of returning empty bundle (wrapping as IllegalStateException is fine).